Fix #701 (long delays to harden CCKD DASD)

RossPatterson opened the pull request:
This PR adds a DASD Hardener thread as discussed at https://github.com/SDL-Hercules-390/hyperion/issues/701#issuecomment-2620060471, to solve the problem of potentially very long delays between guest OS DASD writes and Herules writes to disk.

**Note:** Although the updates to the documentation in this PR recommend settting CCKD DHINT > 0 and CCKD FSYNC=1, this update _does not_ set those as defaults.  I'm open to doing so, but it seemed like defaulting this code to "active" was a big first step.

Fish-Git reviewed (changes requested):
Hi Ross!

First of all, _excellent_ work! &nbsp;`:)`

I like very much the changes you made. They're very simple and straightforward and quite easy to understand and follow! Well done! &nbsp;`:)`

That being said, I can see only three _**extremely** minor_ changes I would like you to make, all in your DASD Hardener Thread (`cckd_dh`) function.

Two of the three have to do with your choice of wording in your comments, believe it or not. _(Yep! Comments are important too!)_:

On line 1730, your comment is "`/* Shadow copy of cckdblk.dhs*/`". I would prefer that be changed to something like "`/* Work copy of cckdblk.dhs  */`" instead. The term "shadow" has particular meaning to cckd code, and I don't wish others to be inadvertently confused by your use of the term "Shadow copy".

The same problem _(but not as egregious)_ occurs on line 1763: I would prefer that the comment "`/* back to the shadows again  */`" be changed to something like "`/* back to seclusion again  */'".

Those are only suggestions though. Feel free to word them however you'd like. I would just prefer you not use the word "shadow", that's all.

So that's 2 of the 3 requested changes.

The 3rd requested change is to a `LOGMSG` being issued instead of a `WRMSG` on line 1825:

```C
    // (Can't use CCKD_TRACE - no dev)
    LOGMSG( "HHC00393I " HHC00393, CCKD_DH_THREAD_NAME, cckdblk.dhint, ctime (&tt_now) );
```
I agree with your comment that  a CCKD_TRACE statement can't be used in this particular case because the message isn't associated with a specific device, but then why use `LOGMSG` the way you are? The end result is a normal message being issued with a normal message number associated to it! So why not use a normal `WRMSG` instead? E.g.

```C
    // "0:0000 %s wait %d seconds at %s"
    WRMSG( HHC00393, "I", CCKD_DH_THREAD_NAME, cckdblk.dhint, ctime( &tt_now ));
```

&nbsp;
That's it! Those are the _ONLY_ things I could find in your code! _(Other than you non-conforming(*) use of braces of course, but I'll let that slide for now. I can always change that at a later time myself should I ever need to make any changes to your code. which, seeing how simple and straightforward it is, is very unlikely to occur!)_.

So.... 3 teensy, tiny, _**extremely** minor_ changes and _POOF!_ Your _greatly_ appreciated contribution to Hercules will be accepted in a flash! &nbsp;`:))`

_(Hey! A code review is a code review is a code review! Right?)_ &nbsp;`;-)`


---
(*) I don't wish to start a brace war, but the convention in Hercules is to place opening braces on the next line, not at the end of the same line. That's not strictly enforced however, but that does seem to be the style most of the rest of Hercules uses.

RossPatterson commented:
> I like very much the changes you made. They're very simple and straightforward and quite easy to understand and follow!

Thanks - my goal was to make it as easy as possible to understand the impact of the changes and to trust them to Do No Harm.

> _(Yep! Comments are important too!)_:

Amen!
 
> On line 1730, your comment is "`/* Shadow copy of cckdblk.dhs*/`". I would prefer that be changed to something like "`/* Work copy of cckdblk.dhs */`" instead. 

Done.
 
> The same problem _(but not as egregious)_ occurs on line 1763: I would prefer that the comment "`/* back to the shadows again */`" be changed to something like "`/* back to seclusion again */'".

Hee hee ... that code was cloned from `cckd_gcol()`, including the comment :-)  I fixed it in both places :-)

> So why not use a normal `WRMSG` instead? 

Done.  I think I went with `LOGMSG` because I started with `CCKD_TRACE`, and then (of course) had to change.

> _(Other than you non-conforming(*) use of braces of course, but I'll let that slide for now. 

FIxed.  There were only 2 cases - the bulk of the code was already using the Hercules style.

> _(Hey! A code review is a code review is a code review! Right?)_  `;-)`

Yup!

Fish-Git reviewed (approved):
Looks good! Thanks! &nbsp;`:)`

Fish-Git commented:
Merged! &nbsp;`:)`
